shared/aws.go

```diff
@@ -0,0 +1,185 @@
+package shared
+
+import (
+	"fmt"
+	"path/filepath"
+	"strconv"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	tf "github.com/gruntwork-io/terratest/modules/terraform"
+	. "github.com/onsi/ginkgo/v2"
+)
+
+type instance struct {
+	region string
+	volume string
+	ami    string
+	class  string
+	subnet string
+	zone   string
+	sg     string
+	key    string
+}
+
+func CreateInstances(qty int64, name string) (ids []string, err error) {
```

Comment:
How are we managing deleting instances created by aws sdk?

Reply:
yes, but thats my next pr for node replace.
sorry lol

Reply:
but something on this path 

```

func DeleteInstance(ip string) error {
	//
	// if nodeType != "server" && nodeType != "agent" {
	// 	return ReturnLogError("invalid node type: %s\n", nodeType)
	// }
	// if name == "" && name != "test-node-replacement" {
	// 	return ReturnLogError("invalid name: %s\n", name)
	// }

	s := session.Must(session.NewSession())
	instances := ec2.New(s, aws.NewConfig().WithRegion("us-east-2"))

	data := &ec2.DescribeInstancesInput{
		Filters: []*ec2.Filter{
			// {
			// 	Name:   aws.String("tag:Name"),
			// 	Values: aws.StringSlice([]string{name}),
			// },
			{
				Name:   aws.String("instance-state-name"),
				Values: aws.StringSlice([]string{"running"}),
			},
			{
				Name:   aws.String("public-ip-address"),
				Values: aws.StringSlice([]string{ip}),
			},
		},
	}

	res, err := instances.DescribeInstances(data)
	if err != nil {
		return ReturnLogError("error describing instances: %w\n", err)
	}

	if len(res.Reservations) > 0 && len(res.Reservations[0].Instances) > 0 {

		instance1 := res.Reservations[0].Instances[0]
		terminateInput := &ec2.TerminateInstancesInput{
			InstanceIds: aws.StringSlice([]string{*instance1.InstanceId}),
		}

		_, err = instances.TerminateInstances(terminateInput)
		if err != nil {
			return ReturnLogError("error terminating instance: %w\n", err)
		}

		LogLevel("info", fmt.Sprintf("\nTerminated\ninstanceName: %+v\ninstanceID: %v",
			*instance1.Tags[0].Value, *instance1.InstanceId))

		return nil
	}

	return nil
}

```